src/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -0,0 +1,123 @@
+{
+    public sealed class DotNetMSBuildSdkResolver : SdkResolver
+    {
+        public override string Name => "Microsoft.DotNet.MSBuildSdkResolver";
+
+        // Default resolver has priority 10000 and we want to go before it and leave room on either side of us. 
+        public override int Priority => 5000;
+
+        public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext context, SdkResultFactory factory)
+        {
+            // These are overrides that are used to force the resolved SDK tasks and targets to come from a given
+            // base directory and report a given version to msbuild (which may be null if unknown. One key use case
+            // for this is to test SDK tasks and targets without deploying them inside the .NET Core SDK.
+            string msbuildSdksDir = Environment.GetEnvironmentVariable("DOTNET_MSBUILD_SDK_RESOLVER_SDKS_DIR");
+            string netcoreSdkVersion = Environment.GetEnvironmentVariable("DOTNET_MSBUILD_SDK_RESOLVER_SDKS_VER");
+
+            if (msbuildSdksDir == null)
+            {
+                string netcoreSdkDir = ResolveNetcoreSdkDirectory(context);
+                if (netcoreSdkDir == null)
+                {
+                    return factory.IndicateFailure(
+                        new[]
+                        {
+                            "Unable to locate the .NET Core SDK. Check that it is installed and that the version"
+                            + "specified in global.json (if any) matches the installed version."
+                        });
+                }
+
+                msbuildSdksDir = Path.Combine(netcoreSdkDir, "Sdks");
+                netcoreSdkVersion = new DirectoryInfo(netcoreSdkDir).Name;;
+            }
+
+            string msbuildSdkDir = Path.Combine(msbuildSdksDir, sdkReference.Name, "Sdk");
+            if (!Directory.Exists(msbuildSdkDir))
+            {
+                return factory.IndicateFailure(
+                    new[] 
+                    {
+                        $"{msbuildSdkDir} not found. Check that a recent enough .NET Core SDK is installed"
+                        + " and/or increase the version specified in global.json. "
+                    });
+            }
+
+            return factory.IndicateSuccess(msbuildSdkDir, netcoreSdkVersion);
+        }
+
+        private string ResolveNetcoreSdkDirectory(SdkResolverContext context)
+        {
+            foreach (string exeDir in GetDotnetExeDirectoryCandidates())
+            {
+                string workingDir = context.SolutionFilePath ?? context.ProjectFilePath;
+                string netcoreSdkDir = Interop.hostfxr_resolve_sdk(exeDir, workingDir);
+
+                if (netcoreSdkDir != null)
+                {
+                    return netcoreSdkDir;
+                }
+            }
+
+            return null;
+        }
+
+        // Search for [ProgramFiles]\dotnet in this order. Only ProgramFiles is defined on 
```

Comment:
Second unfinished sentence should be removed. I put all the detail inline below.